pkgs/tools/misc/vector/default.nix

```diff
@@ -101,7 +101,7 @@ rustPlatform.buildRustPackage {
   # Rust 1.80.0 introduced the unexepcted_cfgs lint, which requires crates to allowlist custom cfg options that they inspect.
   # Upstream is working on fixing this in https://github.com/vectordotdev/vector/pull/20949, but silencing the lint lets us build again until then.
   # TODO remove when upgrading Vector
-  RUSTFLAGS = "--allow unexpected_cfgs";
+  RUSTFLAGS = "--allow unexpected_cfgs --allow dependency_on_unit_never_type_fallback --allow dead_code";
```

Comment:
I believe the `--allow unexpected_cfgs` flag can be removed now, https://github.com/vectordotdev/vector/commit/31dc38f2126d94f1ff7800c82fd0b67cf7e5dfa9 is in vector v0.42.0.